### DISCUSSION TITLE
Cannot display only X values present in the chartEntryModelProducer

### DISCUSSION THREAD

**Discussion by matpascaud** [patrykandpatrick/vico#400] [Questions] [answered]
Hello,

I am struggling to display only X values present in the list of entries.
I have some X values between 1 and 40 and would like to display only the ones present, not all intermediate values.
So in my update function, I added an AxisValueFormatter method but cannot make it work.
This is what I am doing:
`var listEntry = mutableListOf<FloatEntry>()
        val listId = mutableListOf<Int>()

        list.forEach { value->
            val elem = FloatEntry(value.id.toFloat(), value.dbHz)
            listEntry.add(elem)
            listId .add(value.id)
        }
        val horizontalAxisValueFormatter = AxisValueFormatter<AxisPosition.Horizontal.Bottom> { value, _ ->
            if(listId .contains(value.toInt())) {
                String.format("%d", value.toInt())
            } else {
                ""
            }
        }
        if(list.size > 0)
            (chartView.bottomAxis as HorizontalAxis).valueFormatter = horizontalAxisValueFormatter
        tabSkyplotViewModel.chartEntryModelProducer.setEntries(listEntry)`

![Screenshot_20230918-105108](https://github.com/patrykandpatrick/vico/assets/534304/a1fb5059-41b2-465a-b5ab-34478f2a2c79)
I am not using Compose and do not understand what i am doing wrong to avoid display intermediate values.
I would be glad to get any help.

Thank you.

**Comment by patrickmichalik** [comment, marked as answer]
Hello! A `Chart`’s coordinate system is orthogonal. In `ColumnChart`s, columns are positioned according to their _x_ values, so if there are several columns whose _x_ values aren’t evenly spaced out on the real line, the columns won’t be drawn at regular intervals. Thus, you’ll have to normalize the _x_ values and override the `HorizontalAxis` labels as required. A simple solution is as follows.

1. Use consecutive integers, the first one being zero, as the _x_ values.
2. Create a list of labels.
3. In your `AxisValueFormatter`, use the _x_ value passed to the lambda to retrieve the corresponding label: `labels[value.toInt()]`.

**Reply by matpascaud** [reply]
Thank you very much @patrickmichalik, got it working. 
Another remark since I spent much time to find the solution. I used MarkerLabelFormatter to customize the marker.
The name of the function is: MarkerLabelFormatter { markedEntries, chartValues ->
It let me think this method will be called once and it contains markedEntries but in fact it is called for each markedEntry with the updated index. Shouldn't the method take markedEntry as parameter instead of markedEntries?

**Reply by patrickmichalik** [reply]
Multiple `ChartEntry` instances may be highlighted at the same time. For example, in a `LineChart` with several series, several lines are shown, and more than one line can have a point at a given _x_ coordinate. Hence, the type of the parameter is `List<Marker.EntryModel>`, not `Marker.EntryModel`, and the name is `markedEntries`.

**Comment by neobie** [comment]
Hopefully an option true or false will be given to either display the empty intermediate values.

**Reply by Gowsky** [reply]
There is a difference between _x_ values and labels. Entries are displayed in a mathematically accurate way. You should use values that match the desired positioning and possibly format them. The solution mentioned above by @patrickmichalik is straightforward.
